prelude.links

```diff
@@ -257,30 +257,31 @@ fun join(glue,list) {
 # elemBy :: ((a) -> (a) -> Bool) -> (a) -> ([a]) -> Bool
 # notElemBy :: ((a) -> (a) -> Bool) -> (a) -> ([a]) -> Bool
 # span :: ((a) -> Bool) -> ([a]) -> ([a],[a])
+sig span : ((a) ~b~> Bool, [a]) ~b~> ([a], [a])
+fun span(p, xs) {
+  fun go(acc, xs) {
```

Comment:
We should really have an optimisation pass for this; you can eliminate the `go` closure by adding `p` to its parameter list.

Reply:
(very) well-spotted -- done.